arcgis-ios-sdk-samples/Edit data/Edit features (connected)/EditFeaturesOnlineViewController.swift

```diff
@@ -16,58 +16,79 @@ import UIKit
-        self.sketchEditor = AGSSketchEditor()
-        self.mapView.sketchEditor = self.sketchEditor
-        
-        // hide the sketchToolbar initially
-        self.sketchToolbar.isHidden = true
-        
-        // store the feature layer for later use
-        self.featureLayer = featureLayer
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["EditFeaturesOnlineViewController", "FeatureTemplatePickerViewController"]
+        // Hide the sketchToolbar initially.
+        sketchToolbar.isHidden = true
+        // Load the service geodatabase.
+        let damageFeatureService = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer")!
+        loadServiceGeodatabase(from: damageFeatureService)
+    }
+    
+    /// Load and set a service geodatabase from a feature service URL.
+    /// - Parameter serviceURL: The URL to the feature service.
+    func loadServiceGeodatabase(from serviceURL: URL) {
+        let serviceGeodatabase = AGSServiceGeodatabase(url: serviceURL)
+        serviceGeodatabase.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                let featureTable = serviceGeodatabase.table(withLayerID: 0)!
+                self.featureTable = featureTable
+                self.serviceGeodatabase = serviceGeodatabase
+                // Add the feature layer to the operational layers on map.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Store the feature layer for later use.
+                self.featureLayer = featureLayer
+                self.mapView.map?.operationalLayers.add(featureLayer)
+                self.mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -9184518.55, y: 3240636.90, spatialReference: .webMercator()), scale: 7e5))
+            }
+        }
     }
     
     private func dismissFeatureTemplatePickerViewController() {
         self.dismiss(animated: true)
     }
     
+    /// Apply local edits to the geodatabase.
     func applyEdits() {
-        // show progress hud
-        UIApplication.shared.showProgressHUD(message: "Applying edits")
-        
-        (featureLayer.featureTable as! AGSServiceFeatureTable).applyEdits { [weak self] (_, error) in
-            UIApplication.shared.hideProgressHUD()
-            
-            if let error = error {
-                self?.presentAlert(message: "Error while applying edits: \(error.localizedDescription)")
-            } else {
-                self?.presentAlert(message: "Edits applied successfully!")
+        if serviceGeodatabase.hasLocalEdits() {
+            serviceGeodatabase.applyEdits { [weak self] (featureTableEditResults: [AGSFeatureTableEditResult]?, error: Error?) in
```

Comment:
Does the README need to be updated since we are now applying edits on the service geodatabase?

Reply:
Yes, it will be done in a 3rd PR. I'll first update the designs internally before making changes to public PRs. Internal number: `common-samples/pull/3380`